stdlib/public/Platform/tgmath.swift.gyb

```diff
@@ -59,6 +59,88 @@ public func trunc<T: FloatingPoint>(_ x: T) -> T {
   return x.rounded(.towardZero)
 }
 
+@_transparent
+public func scalbn<T: FloatingPoint>(_ x: T, _ n : Int) -> T {
+  return T(sign: .plus, exponent: T.Exponent(n), significand: x)
+}
+
+@_transparent
+public func modf<T: FloatingPoint>(_ x: T) -> (T, T) {
+  // inf/NaN: return canonicalized x, fractional part zero.
+  guard x.isFinite else { return (x+0, 0) }
+  let integral = trunc(x)
+  let fractional = x - integral
+  return (integral, fractional)
+}
+
+@_transparent
+public func frexp<T: BinaryFloatingPoint>(_ x: T) -> (T, Int) {
+  guard x.isFinite else { return (x+0, 0) }
+  guard x != 0 else { return (x, 0) }
+  // The C stdlib `frexp` uses a different notion of significand / exponent
+  // than IEEE 754, so we need to adjust them by a factor of two.
+  return (x.significand / 2, Int(x.exponent + 1))
+}
+
+%for T in ['Float','Double']:
+@available(swift, obsoleted: 4.2)
+@_transparent
+public func scalbn(_ x: ${T}, _ n : Int) -> ${T} {
+  return ${T}(sign: .plus, exponent: n, significand: x)
+}
+
+@available(swift, obsoleted: 4.2)
+@_transparent
+public func modf(_ x: ${T}) -> (${T}, ${T}) {
+  // inf/NaN: return canonicalized x, fractional part zero.
+  guard x.isFinite else { return (x+0, 0) }
+  let integral = trunc(x)
+  let fractional = x - integral
+  return (integral, fractional)
+}
+
+@available(swift, obsoleted: 4.2)
+@_transparent
+public func frexp(_ x: ${T}) -> (${T}, Int) {
+  guard x.isFinite else { return (x+0, 0) }
+  guard x != 0 else { return (x, 0) }
+  return (x.significand / 2, Int(x.exponent + 1))
+}
+
+@available(swift, deprecated: 4.2, message: "Use scalbn<T>(:T,:Int).")
```

Comment:
Does `scalbn` give the same result? If so, then maybe a `renamed:` instead of `message:` would provide a better developer experience (i.e. fix-it and migration).

**UPD**: It does look like the implementation for `scalbn` is similar.

Reply:
Yeah, they're identical in actual behavior for every current type. They would behave differently for decimal types, but we don't really want these to be the interfaces going forward anyway.